### PR TITLE
added main logic for working with web

### PR DESCRIPTION
**Pull request by ArtemReizer**

### CODE REVIEW

File: src/main/java/com/app/bookstore/config/MapperConfig.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package com.app.bookstore.config;
+
+import org.mapstruct.InjectionStrategy;
+import org.mapstruct.NullValueCheckStrategy;
+
+@org.mapstruct.MapperConfig(
```
Comment (IvanFadieiev) [inline src/main/java/com/app/bookstore/config/MapperConfig.java]: If we name this class as "MapperConfiguration" we can use @MapperConfig annotation

File: src/main/java/com/app/bookstore/repository/impl/BookRepositoryImpl.java [hunk 2]
```diff
@@ -33,7 +29,7 @@ public Book save(Book book) {
             if (transaction != null) {
                 transaction.rollback();
             }
-            throw new HibernateException("Can't save the book: " + book, e);
+            throw new EntityNotFoundException("Can't save the book: " + book, e);
```
Comment (IvanFadieiev) [inline src/main/java/com/app/bookstore/repository/impl/BookRepositoryImpl.java]: I think that this exception it's not accurate for this case, better create a new DataProcessingException for example

File: src/main/java/com/app/bookstore/repository/impl/BookRepositoryImpl.java [hunk 3]
```diff
@@ -46,7 +42,16 @@ public List<Book> findAll() {
...
+            throw new EntityNotFoundException("Can't get all books", e);
+        }
+    }
+
+    @Override
+    public Book getById(Long id) {
+        try (Session session = sessionFactory.openSession()) {
+            return session.get(Book.class, id);
+        } catch (Exception e) {
+            throw new EntityNotFoundException("Cannot get book by id: " + id, e);
```
Comment (IvanFadieiev) [inline src/main/java/com/app/bookstore/repository/impl/BookRepositoryImpl.java]: It's better to return Optional here and handle it on service layer

**Review by IvanFadieiev — CHANGES_REQUESTED** [review]
Great job! Some small details to change :)

**Review by IvanFadieiev — APPROVED** [review]
Great job!

File: pom.xml [hunk 4]
```diff
@@ -5,8 +5,8 @@
 	<parent>
 		<groupId>org.springframework.boot</groupId>
 		<artifactId>spring-boot-starter-parent</artifactId>
-		<version>3.1.3</version>
-		<relativePath/> <!-- lookup parent from repository -->
+		<version>3.1.2</version>
```
Comment (vovasalyha) [inline pom.xml:8]: was that necessary to downgrade version of spring boot?

**Review by vovasalyha — APPROVED** [review]
GJ

**Pull request merged**